src/main/kotlin/cn/yiiguxing/plugin/translate/trans/deepl/DeeplTranslator.kt

```diff
@@ -109,27 +109,37 @@ object DeeplTranslator : AbstractTranslator(), DocumentationTranslator {
     }
 
     override fun doTranslate(text: String, srcLang: Lang, targetLang: Lang): Translation {
-        return SimpleTranslateClient(this, DeeplTranslator::call, DeeplTranslator::parseTranslation).execute(
-            text,
-            srcLang,
-            targetLang
-        )
+        return SimpleTranslateClient(
+            this,
+            { _, _, _ ->call(text, srcLang, targetLang, false) },
+            DeeplTranslator::parseTranslation
+        ).execute(text, srcLang, targetLang)
     }
 
-    private fun call(text: String, srcLang: Lang, targetLang: Lang): String {
+    private fun call(text: String, srcLang: Lang, targetLang: Lang, isDocument: Boolean): String {
         val settings = Settings.deeplTranslateSettings
         val privateKey = settings.getAppKey()
         val splitPrivatekeys = privateKey.split(":")
         val isFree = (splitPrivatekeys.size == 2 && splitPrivatekeys[1] == "fx")
         val requestURL = if (isFree) DEEPL_FREE_TRANSLATE_API_URL else DEEPL_PRO_TRANSLATE_API_URL
-
-        return Http.post(
-            requestURL,
+        val postData: LinkedHashMap<String, String> = linkedMapOf(
             "auth_key" to privateKey,
-            "tag_handling" to "xml",
             "target_lang" to targetLang.deeplLanguageCode,
             "text" to text
         )
+
+        if (isDocument) {
+            postData["tag_handling"] = "html"
+        }
+
+        if (srcLang !== Lang.AUTO) {
+            postData["source_lang"] = srcLang.langName
```

Comment:
Should be `postData["source_lang"] = srcLang.deeplLanguageCode` here.

Reply:
Thanks. Corrected.